### PR TITLE
Add YAML snippet to `json.JSONDataSet` to class

### PR DESCRIPTION
**Pull request by punitvara**
## Description
Adds yaml examples in dataset docs as discussed in #1155

## Development notes
Added yaml docs for the following datasets.
json.JSONDataSet

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by datajoely** [comment]
Hi @punitvara nice work - I've clarified the  description. You may also want to include a simple local file example rather than jumping to a GCS one.

Two points on process - as part of the Linux Foundation you need to make the DCO sign-off module happy. The instructions for [this are in PR](https://github.com/kedro-org/kedro/pull/1177/files) . Once you mark the PR ready for review the core team will review and ensure this gets in :)

**Review by datajoely — APPROVED** [review]

**Review by SajidAlamQB — APPROVED** [review]

**Comment by datajoely** [comment]
@SajidAlamQB  any idea why CI isn't passing?

**Comment by punitvara** [comment]
I am also curious about ci failures. From errors its not clear. Usually I check in this kind of cases if I have rebased new branch on main but that is also fine.

**Comment by SajidAlamQB** [comment]
> @SajidAlamQB any idea why CI isn't passing?

Not exactly sure but @AntonyMilneQB was looking into it yesterday.

File: RELEASE.md
Comment (AntonyMilneQB) [inline RELEASE.md:58]:
```suggestion
```
0.17.6 was the last version we did this for - now we just just rely on the [automatic GitHub feature](https://github.com/kedro-org/kedro/graphs/contributors) to track contributors 🙂

**Review by AntonyMilneQB — APPROVED** [review]
Thank you very much for this @punitvara 🎉  Don't worry about the CI failures - we're working on a fix for it and will make sure this gets merged in when we've done that.

**Comment by punitvara** [comment]
ok Thanks. @AntonyMilneQB  Nothing else is needed for this right ? Then I would pick something else to work on.

**Review by idanov — APPROVED** [review]

**Comment by AntonyMilneQB** [comment]
> ok Thanks. @AntonyMilneQB Nothing else is needed for this right ? Then I would pick something else to work on.

Right 👍

**Pull request merged**
